taxcalc/records.py

```diff
@@ -661,3 +656,30 @@ def read(self, data):
         self.e60100 = self.p60100
         self.e27860 = self.s27860
         self.SOIYR = np.repeat(2008, self.dim)
+
+    def mutate_imputations(self):
+        self._cmbtp_itemizer = imputation(self.e17500, self.e00100,
+            self.e18400, self.e18425, self.e62100, self.e00700,
+            self.e04470, self.e21040, self.e18500, self.e20800)
+
+
+@vectorize([float64(float64, float64, float64, float64, float64, float64,
+            float64, float64, float64, float64)])
+def imputation(e17500, e00100, e18400, e18425, e62100, e00700, e04470,
+               e21040, e18500, e20800):
+
+    """
+    Calculates _cmbtp_itemizer
+    Uses vectorize decorator to speed-up calculation process
+    with NumPy data structures
+    """
+
+    # temp variables to make it easier to read
+    x = max(0., e17500 - max(0., e00100) * 0.075)
+    y = -1 * min(x, 0.025 * max(0., e00100))
+    z = max(0, max(e18400, e18425))
+
+    _cmbtp_itemizer = (y + e62100 + e00700 + e04470 + e21040 - z - e00100
```

Comment:
Also, it probably makes more sense to get rid of the -1 coefficient on y and then subtract y right before we subtract z rather than add y (with the -1 coef) at the beginning of this expression.